parsers/US_PJM.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+
+"""Parser for the PJM area of the United States."""
+
+import arrow
+from bs4 import BeautifulSoup
+import demjson
+import re
+import requests
+
+
+url = 'http://www.pjm.com/markets-and-operations.aspx'
+
+mapping = {
+            'Coal': 'coal',
+            'Gas': 'gas',
+            'Hydro': 'hydro',
+            'Multiple Fuels': 'unknown',
+            'Nuclear': 'nuclear',
+            'Oil': 'oil',
+            'Other': 'unknown',
+            'Other Renewables': 'unknown',
+            'Solar': 'solar',
+            'Wind': 'wind'
+            }
+
+
+def extract_data(session = None):
+    """
+    Makes a request to the PJM data url.
+    Finds timestamp of current data and converts into a useful form.
+    Finds generation data inside script tag.
+    Returns a tuple of generation data and datetime.
+    """
+
+    s = session or requests.Session()
+    req = requests.get(url)
+    soup = BeautifulSoup(req.content, 'html.parser')
+
+    try:
+        time_div = soup.find("div", id="asOfDate").text
+    except AttributeError:
+        raise LookupError('No data is available for US-PJM.')
+
+    time_pattern = re.compile(r"""(\d{1,2}     #Hour can be 1/2 digits.
+                                   :           #Separator.
+                                   \d{2})\s    #Minutes must be 2 digits with a space after.
+                                   (a.m.|p.m.) #Either am or pm allowed.""", re.X)
+
+    latest_time = re.search(time_pattern, time_div)
+
+    time_data = latest_time.group(1)
+    am_or_pm = latest_time.group(2)
+    hour = int(time_data.split(":")[0])
+    minute = int(time_data.split(":")[1])
+
+    #Time format used by PJM is slightly unusual and needs to be converted so arrow can use it.
+    if am_or_pm == "p.m." and hour != 12:
+        hour += 12
```

Comment:
Do you know what `hour` is at midnight? 0 or 12?

Reply:
That is a very good point I'll check.